### ISSUE TITLE
WASM - Int32 Math and conversion traps

### ISSUE THREAD

**Issue by Cellule** [Microsoft/ChakraCore#1878]
Verify and implement math operators that need to trap
https://github.com/WebAssembly/design/blob/binary_0xc/AstSemantics.md#32-bit-integer-operators

**Comment by Cellule** [comment]
@arun-etm @Krovatkin @commonlisp I seem to recall one of you guys was working on this. Can someone confirm ?

**Comment by Krovatkin** [comment]
yes! i'm working on this one!

**Comment by Krovatkin** [comment]
@Cellule A quick question. Should this item cover traps for i64 instructions too?  
I added exceptions to the following i32 instructions. 

```
i32.trunc_s/f32   
i32.trunc_u/f32   
i32.trunc_s/f64   
i32.trunc_u/f64   
i32.div_s         
i32.rem_s         
i32.div_u         
i32.rem_u   
```

This list seems to include every arithmetic exception I could scrub from ml-proto.

**Comment by Cellule** [comment]
@Krovatkin Yes I think this item should cover int64 as well.
You can make a separate PR for it however.

**Comment by Krovatkin** [comment]
@Cellule  Do you think if we could maybe 

* For i64.trunc_XX add to issues' descriptions (currently unimplemented and unassigned) that these instructions need to handle overflow and invalid conversions
* For i64.div/rem, open a new issue to "add checks to handle DivideByZero, IntegerOverflow"? 
* Rename this one into Math traps for i32 instructions

This way if someone else has a few free cycles we can work in parallel. I'm planning to start on  `i64.trunc_XX` tomorrow morning unless you and @MikeHolman completely dislike https://github.com/Microsoft/ChakraCore/pull/1848 in which case I'll need to spend more time on that one.

**Comment by Cellule** [comment]
If you prefer it that way we can do that.

**Comment by Cellule** [comment]
I created #1900 and updated the related issues for int64 truncation from float

**Comment by Krovatkin** [comment]
@Cellule Thank you! This is mostly "just in case"; I don't want to become a bottleneck :-)
